src/components/SingleArticle.jsx

```diff
@@ -22,6 +24,21 @@ const SingleArticle = () => {
       });
   }, [article_id]);
 
+  const handleVote = (voteChange) => {
+    if (!article) return;
```

Comment:
I am not sure why this would need to be checked here. I would imagine if there is no article either it is loading, which you are only displaying a p tag for or there is an error, which is also displaying something other than the article, so this function should not be reachable if there is no article.